file/writable_file_writer.cc

```diff
@@ -814,12 +828,13 @@ IOStatus WritableFileWriter::WriteDirectWithChecksum(
   // TODO: need to be improved since it sort of defeats the purpose of the rate
   // limiter
   size_t data_size = left;
-  if (rate_limiter_ != nullptr) {
+  Env::IOPriority rate_limiter_priority_used =
```

Comment:
One consequence of bypassing rate-limiter's charging instead of charging it with the Env::IO_TOTAL is that we need to decide `rate_limiter_priority_used` before `size = rate_limiter_->RequestToken`. 

It's not a big deal as long as the same `writable_file_`'s `io_priority_` is not shared between threads. Otherwise 
we might have `writable_file_`'s `io_priority_` being changed between the loop `while (data_size > 0) {}` below.

I don't think this is the case and I am double checking with Anand.

Reply:
Hey @anand1976 , another quick question related to concurrency of FileSystem. 

Is it guaranteed that `writable_file_`'s `io_priority_` is not shared between threads? I don't think so by briefly inspecting the code but figured I should double check with you.

If you need more context for my question, see the PR comment right above.

Reply:
Yes. We never have multiple threads writing to the same ```WritableFile```.